evm/src/main/java/org/hyperledger/besu/evm/fluent/SimpleWorld.java

```diff
@@ -107,11 +120,16 @@ public void revert() {
 
   @Override
   public void commit() {
-    parent.accounts.putAll(accounts);
+    accounts.forEach(
+        (address, account) -> {
+          if (!account.commit()) {
+            accounts.put(address, account);
+          }
+        });
```

Comment:
this is peculiar - if we cannot commit the account, we re-add it back to accounts?  should this be `parent.accounts.put(address,account)` ?

Reply:
Also yes, the parent.
SimpleWorld lacks deep testing and is meant for "toy" environments.  I would like to run general world state reference tests against the EVM lib alone, but there would be some ref tests duplication.  And we need a fake data store substrate, which is what this fills.  Real testing will be when ref tests pass.